### DISCUSSION TITLE
Create custom template?

### DISCUSSION THREAD

**Discussion by ntapdave** [NetApp/harvest#637] [Q&A] [answered]
Hi Harvest team, I'm so happy to see Harvest updated like this, but I've found a use case that's not currently covered- I'd like to create a template that collects a smaller number of metrics more frequently. I use this in lab environments for detailed performance testing and I need 1- or 5-second granularity. I don't need 90% of the metrics that are collected by default and reducing the number of metrics significantly reduces the overhead of collecting at that rate. In the previous version of Harvest I could create a custom template and associate the poller with it. 

I'm assuming I can build a custom container by editing out the metrics I don't need in the default templates, is there another way to insert a completely custom collector template?

**Comment by ntapdave** [comment]
Note, I'm running harvest in a container so instead of building a custom container, I'm going to try running the container with a modified template directory bound to the container to overlay the default one. That should work and allow continued use of the default container.

**Comment by ntapdave** [comment]
And now that I RTFM I see the option to specify a custom template file which means I don't have to overlay the default template, just map in the custom one.

**Comment by rahulguptajss** [comment, marked as answer]
@ntapdave We have made changes to custom template functionality in our upcoming release. More info @ https://github.com/NetApp/harvest/tree/main/conf#extend-an-existing-object-template
For your use case, you can remove the mapping from default.yaml and keep your custom template mapping in custom.yaml.
